tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs

```diff
@@ -721,6 +720,217 @@ void TestCase(ClientEncryption testCaseClientEncription, BsonDocument masterKey,
+                    keyVaultMongoClient: null,
+                    expectedNumberOfClients: 2,
+                    clientEncryptedEventsExpectation:
+                    new[]
+                    {
+                        "listCollections;db",
+                        "find;keyvault",
+                        "insert;db",
+                        "find;db"
+                    },
+                    clientKeyVaultEventsExpectation: null),
+                CaseWithAsync(
+                    name: "case 2",
+                    maxPoolSize: 1,
+                    bypassAutoEncryption: false,
+                    keyVaultMongoClient: "client_keyvault",
+                    expectedNumberOfClients: 2,
+                    clientEncryptedEventsExpectation:
+                    new[]
+                    {
+                        "listCollections;db",
+                        "insert;db",
+                        "find;db"
+                    },
+                    clientKeyVaultEventsExpectation:
+                    new []
+                    {
+                        "find;keyvault"
+                    }),
+                CaseWithAsync(
+                    name: "case 3",
+                    maxPoolSize: 1,
+                    bypassAutoEncryption: true,
+                    keyVaultMongoClient: null,
+                    expectedNumberOfClients: 2,
+                    clientEncryptedEventsExpectation:
+                    new[]
+                    {
+                        "find;db",
+                        "find;keyvault"
+                    },
+                    clientKeyVaultEventsExpectation: null),
+                CaseWithAsync(
+                    name: "case 4",
+                    maxPoolSize: 1,
+                    bypassAutoEncryption: true,
+                    keyVaultMongoClient: "client_keyvault",
+                    expectedNumberOfClients: 1,
+                    clientEncryptedEventsExpectation:
+                    new[]
+                    {
+                        "find;db",
+                    },
+                    clientKeyVaultEventsExpectation:
+                    new []
+                    {
+                        "find;keyvault"
+                    }),
+
+                // cases 5-8 use "MaxPoolSize: 0" which is not supported by the c# driver
+            };
+
+            foreach (var asyncCases in testCases)
+            {
+                foreach (var asyncCase in asyncCases)
+                {
+                    yield return asyncCase;
+                }
+            }
+
+            IEnumerable<object[]> CaseWithAsync(
+                string name,
+                int maxPoolSize,
+                bool bypassAutoEncryption,
+                string keyVaultMongoClient,
+                int expectedNumberOfClients,
+                string[] clientEncryptedEventsExpectation,
+                string[] clientKeyVaultEventsExpectation)
```

Comment:
Instead of splitting strings you can use tuples here:
```suggestion
                (string, string)[] clientEncryptedEventsExpectation,
                (string, string)[] clientKeyVaultEventsExpectation)
```

Reply:
answered above